packages/zoe/src/contractFacet.js

```diff
@@ -444,13 +323,11 @@ export function buildRootObject(_vatPowers) {
         params.inviteIssuer,
         params.zoeForZcf,
       );
+      /** @type {Promise<Invite<any>>} */
       const inviteP = E(contractCode).makeContract(contractFacet);
       return {
         inviteP,
-        zcfForZoe: makeZcfForZoe(
-          params.instanceData.instanceHandle,
-          params.zoeForZcf,
-        ),
+        zcfForZoe: makeZcfForZoe(params.instanceData.handle, params.zoeForZcf),
```

Comment:
There is some pretty serious confusion about the public InstanceRecord (`handle` property) versus what I'm now calling the InternalInstanceData (`instanceHandle` property).

The table.js implementation always adds a `handle` property to records it creates, and the public Zoe/Zcf APIs expose InstanceRecords with that property.  Any `instanceHandle` is filtered out when we do so.

But, internal between Zoe and Zcf, we have `instanceHandle` instead of `handle`.  That difference seems gratuitous, and we should decide what to do about it, since at least `mintAndSellNFT.js` depends on having access to a field of the InstanceRecord called `handle`.

Reply:
> which is where the name of the key is declared.

That key name is just for debugging the WeakMap.  It isn't actually used in the record.

Reply:
All the tests I could find destructure the results of `getInstanceRecord()` and reference the `handle` property.

Reply:
in [`zoe/test/swingsetTests/zoe/vat-bob.js`,](https://github.com/Agoric/agoric-sdk/blob/8078ef07fbe9036a84b45bdb3facb5c33cf3fc71/packages/zoe/test/swingsetTests/zoe/vat-bob.js#L511) I see
``` js
      const { installationHandle, issuerKeywordRecord, terms } = await E(
        zoe,
      ).getInstanceRecord(inviteValue[0].instanceHandle);
```

in [`zoe/test/unitTests/contracts/tests-atomicSwap.js`](https://github.com/Agoric/agoric-sdk/blob/8078ef07fbe9036a84b45bdb3facb5c33cf3fc71/packages/zoe/test/unitTests/contracts/test-atomicSwap.js#L79), I see
``` js
  const {
    installationHandle: bobInstallationId,
    issuerKeywordRecord: bobIssuers,
  } = zoe.getInstanceRecord(bobInviteValue.instanceHandle);
```

What am I missing?

Reply:
Oh, wait. I was supposed to be looking for the instanceHandle there.  Woops.  Thanks for catching me on that.

Reply:
The `instanceHandle` in `instanceData` should be renamed to `handle`. I'll bet there are other cascading changes. Do you want to make those changes in this PR (I could take care of that if you like) or I could do it as a separate thing.

Reply:
> But, internal between Zoe and Zcf, we have instanceHandle instead of handle. That difference seems gratuitous, and we should decide what to do about it, since at least mintAndSellNFT.js depends on having access to a field of the InstanceRecord called handle.

Yes, the difference is gratuitous and should go away.  `instanceHandle` should be renamed to `handle` as it's put in the `instanceData` record, and should retain that name in the instanceRecord passed around within contractFacet. I think the only change is that `visibleInstanceRecordFields()` should expect `handle` rather than `instanceHandle`. 

Sorry for the confusion I added. I think it's a result of Kate's asking me to document the debugging names of the key fields in state.js, which made me think the keys had those names on retrieval, which I now see was erroneous.

Once I got that in my head, I seem to have been blind to evidence to the contrary.

Reply:
Naming the `makeTable` parameter `key` was misleading.  So, I've changed it to `keyDebugName`, and defaulted it to `Handle`.

Don't worry about misunderstanding the functionality (I've made several such mistakes in the process of creating this PR): that's exactly the kind of thing that adding types helps shed light on, so consider this a useful learning experience for all of us.